### PR TITLE
feat(trusted): add bounce vm on azure

### PR DESCRIPTION
**Pull request by smerle33**
as per https://github.com/jenkins-infra/helpdesk/issues/3486

create a bounce VM on azure 

from https://github.com/jenkins-infra/azure/pull/331

### CODE REVIEW

**Comment by smerle33** [comment]
https://learn.microsoft.com/en-us/azure/virtual-network/ip-services/public-ip-addresses EOL for Basic SKU for public ip in 2025

File: trusted.ci.jenkins.io.tf [hunk 1]
```diff
@@ -1,8 +1,50 @@
+####################################################################################
+## TRUSTED NETWORK Define in https://github.com/jenkins-infra/azure-net/blob/main/vnets.tf
+####################################################################################
```
Comment (dduportal) [inline trusted.ci.jenkins.io.tf:3]:
```suggestion
# Network resources defined in https://github.com/jenkins-infra/azure-net
```

File: trusted.ci.jenkins.io.tf [hunk 2]
```diff
@@ -1,8 +1,50 @@
...
+data "azurerm_subnet" "trusted_permanent_agents" {
+  name                 = "${data.azurerm_virtual_network.trusted.name}-trusted-jenkins-ci-io-permanent-agents"
+  virtual_network_name = data.azurerm_virtual_network.trusted.name
+  resource_group_name  = data.azurerm_resource_group.trusted.name
+}
+
+####################################################################################
+## TRUSTED RESOURCES
+####################################################################################
+# Resources groups for TRUSTED Agents
```
Comment (dduportal) [inline trusted.ci.jenkins.io.tf:30]:
```suggestion
```

File: trusted.ci.jenkins.io.tf [hunk 3]
```diff
@@ -1,8 +1,50 @@
...
   name     = "jenkinsinfra-trustedvmagents"
-  location = "East US"
+  location = var.location
+  tags     = local.default_tags
+}
+resource "azurerm_resource_group" "trusted_ci_jenkins_io_permanent_agents" {
+  name     = "jenkinsinfra-trusted-permanent-vmagents"
+  location = var.location
+  tags     = local.default_tags
+}
```
Comment (dduportal) [inline trusted.ci.jenkins.io.tf:40]:
```suggestion
```

Comment (dduportal) [reply trusted.ci.jenkins.io.tf:40]: Pair review: off scope

File: trusted.ci.jenkins.io.tf [hunk 4]
```diff
@@ -50,3 +92,64 @@ resource "azurerm_role_assignment" "trusted_ci_jenkins_io_allow_packer" {
...
+############################# VMs ####################################
+
+# NETWORKING FROM azure-net repository
+# check in vnets.tf
+#
+
+# BOUNCE VM
+## PUBLIC IP
+resource "azurerm_public_ip" "trusted_bounce" {
+  name                = "trusted-bounce-external-ip"
```
Comment (dduportal) [inline trusted.ci.jenkins.io.tf:105]:
```suggestion
  name                = "trusted-bounce"
```

File: trusted.ci.jenkins.io.tf [hunk 5]
```diff
@@ -50,3 +92,64 @@ resource "azurerm_role_assignment" "trusted_ci_jenkins_io_allow_packer" {
...
+  location            = azurerm_resource_group.trusted_ci_jenkins_io_controller.location
+  resource_group_name = azurerm_resource_group.trusted_ci_jenkins_io_controller.name
+  allocation_method   = "Static"
+  sku                 = "Standard"
+  tags                = local.default_tags
+}
+## NETWORK INTERFACE with public ip and internal ip
+resource "azurerm_network_interface" "trusted_bounce" {
+  name                = "trusted-bounce-internal-nic"
```
Comment (dduportal) [inline trusted.ci.jenkins.io.tf:114]:
```suggestion
  name                = "trusted-bounce"
```

File: trusted.ci.jenkins.io.tf (hunk 4 above)
Comment (dduportal) [reply trusted.ci.jenkins.io.tf:105]: Pair review: naming

File: trusted.ci.jenkins.io.tf (hunk 5 above)
Comment (dduportal) [reply trusted.ci.jenkins.io.tf:114]: Pair review: naming

File: trusted.ci.jenkins.io.tf [hunk 6]
```diff
@@ -50,3 +92,64 @@ resource "azurerm_role_assignment" "trusted_ci_jenkins_io_allow_packer" {
...
+  location                        = azurerm_resource_group.trusted_ci_jenkins_io_controller.location
+  size                            = "Standard_B1s"
+  admin_username                  = "adminuser"
+  disable_password_authentication = true
+  network_interface_ids = [
+    azurerm_network_interface.trusted_bounce.id,
+  ]
+
+  admin_ssh_key {
+    username   = "adminuser"
```
Comment (dduportal) [inline trusted.ci.jenkins.io.tf:139]: Same as https://github.com/jenkins-infra/azure/pull/333/files#r1176304473

File: trusted.ci.jenkins.io.tf [hunk 7]
```diff
@@ -50,3 +92,64 @@ resource "azurerm_role_assignment" "trusted_ci_jenkins_io_allow_packer" {
...
+    subnet_id                     = data.azurerm_subnet.trusted_controller.id
+  }
+}
+## MACHINE (bounce)
+resource "azurerm_linux_virtual_machine" "trusted_bounce" {
+  name                            = "trusted-bounce"
+  resource_group_name             = azurerm_resource_group.trusted_ci_jenkins_io_controller.name
+  location                        = azurerm_resource_group.trusted_ci_jenkins_io_controller.location
+  size                            = "Standard_B1s"
+  admin_username                  = "adminuser"
```
[the remark this answers is not in the document]
Comment (smerle33) [reply trusted.ci.jenkins.io.tf:132]: https://github.com/jenkins-infra/azure-net/blob/29fd6351f53184bb7d851c332ef2a0550cb39c20/vpn.tf#L119

File: trusted.ci.jenkins.io.tf [hunk 8]
```diff
@@ -50,3 +92,64 @@ resource "azurerm_role_assignment" "trusted_ci_jenkins_io_allow_packer" {
...
+  disable_password_authentication = true
+  network_interface_ids = [
+    azurerm_network_interface.trusted_bounce.id,
+  ]
+
+  admin_ssh_key {
+    username   = "adminuser"
+    public_key = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQC5K7Ro7jBl5Kc68RdzG6EXHstIBFSxO5Da8SQJSMeCbb4cHTYuBBH8jNsAFcnkN64kEu+YhmlxaWEVEIrPgfGfs13ZL7v9p+Nt76tsz6gnVdAy2zCz607pAWe7p4bBn6T9zdZcBSnvjawO+8t/5ue4ngcfAjanN5OsOgLeD6yqVyP8YTERjW78jvp2TFrIYmgWMI5ES1ln32PQmRZwc1eAOsyGJW/YIBdOxaSkZ41qUvb9b3dCorGuCovpSK2EeNphjLPpVX/NRpVY4YlDqAcTCdLdDrEeVqkiA/VDCYNhudZTDa8f1iHwBE/GEtlKmoO6dxJ5LAkRk3RIVHYrmI6XXSw5l0tHhW5D12MNwzUfDxQEzBpGK5iSfOBt5zJ5OiI9ftnsq/GV7vCXfvMVGDLUC551P5/s/wM70QmHwhlGQNLNeJxRTvd6tL11bof3K+29ivFYUmpU17iVxYOWhkNY86WyngHU6Ux0zaczF3H6H0tpg1Ca/cFO428AVPw/RTJpcAe6OVKq5zwARNApQ/p6fJKUAdXap+PpQGZlQhPLkUbwtFXGTrpX9ePTcdzryCYjgrZouvy4ZMzruJiIbFUH8mRY3xVREVaIsJakruvgw3b14oQgcB4BwYVBBqi62xIvbRzAv7Su9t2jK6OR2z3sM/hLJRqIJ5oILMORa7XqrQ== smerle@MacBook-Pro-de-Stephane.local"
```
Comment (dduportal) [inline trusted.ci.jenkins.io.tf:140]: - Change to ECDSA (if it works) ideally
- Don't add the suffix ;)

File: trusted.ci.jenkins.io.tf [hunk 9]
```diff
@@ -50,3 +92,64 @@ resource "azurerm_role_assignment" "trusted_ci_jenkins_io_allow_packer" {
...
+  }
+
+  os_disk {
+    caching              = "ReadWrite"
+    storage_account_type = "Standard_LRS"
```
Comment (dduportal) [inline trusted.ci.jenkins.io.tf:145]:
```suggestion
    storage_account_type = "Standard_LRS" # Use HDD (cheaper than SSD) as this machine does not need performances
```

File: trusted.ci.jenkins.io.tf [hunk 10]
```diff
@@ -50,3 +92,64 @@ resource "azurerm_role_assignment" "trusted_ci_jenkins_io_allow_packer" {
...
+    disk_size_gb         = 32 #minimal size for ubuntu 22.04 image
```
Comment (dduportal) [inline trusted.ci.jenkins.io.tf:146]:
```suggestion
    disk_size_gb         = 32 # Minimal size for ubuntu 22.04 image
```

**Review by dduportal — COMMENTED** [review]
Pair review 1: added a set of comment together, let's update this content to scope only to the bounce VM.

File: trusted.ci.jenkins.io.tf (hunk 8 above)
Comment (smerle33) [reply trusted.ci.jenkins.io.tf:140]: `the provided ecdsa-sha2-nistp521 SSH key is not supported. Only RSA SSH keys are supported by Azure`

File: locals.tf [hunk 11]
```diff
@@ -46,4 +46,9 @@ locals {
     scope      = "terraform-managed"
     repository = "jenkins-infra/azure"
   }
+
+  trusted = {
```
Comment (dduportal) [inline locals.tf:50]: => `trusted.ci.jenkins.io` 

- Avoid confusion with the "trusted" adjective
- full service name helps to map with the file `trusted.ci.jenkins.io.tf`

File: locals.tf [hunk 12]
```diff
@@ -46,4 +46,9 @@ locals {
...
+    admin_username   = "jenkins-infra-team"
+    admin_public_key = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQC5K7Ro7jBl5Kc68RdzG6EXHstIBFSxO5Da8SQJSMeCbb4cHTYuBBH8jNsAFcnkN64kEu+YhmlxaWEVEIrPgfGfs13ZL7v9p+Nt76tsz6gnVdAy2zCz607pAWe7p4bBn6T9zdZcBSnvjawO+8t/5ue4ngcfAjanN5OsOgLeD6yqVyP8YTERjW78jvp2TFrIYmgWMI5ES1ln32PQmRZwc1eAOsyGJW/YIBdOxaSkZ41qUvb9b3dCorGuCovpSK2EeNphjLPpVX/NRpVY4YlDqAcTCdLdDrEeVqkiA/VDCYNhudZTDa8f1iHwBE/GEtlKmoO6dxJ5LAkRk3RIVHYrmI6XXSw5l0tHhW5D12MNwzUfDxQEzBpGK5iSfOBt5zJ5OiI9ftnsq/GV7vCXfvMVGDLUC551P5/s/wM70QmHwhlGQNLNeJxRTvd6tL11bof3K+29ivFYUmpU17iVxYOWhkNY86WyngHU6Ux0zaczF3H6H0tpg1Ca/cFO428AVPw/RTJpcAe6OVKq5zwARNApQ/p6fJKUAdXap+PpQGZlQhPLkUbwtFXGTrpX9ePTcdzryCYjgrZouvy4ZMzruJiIbFUH8mRY3xVREVaIsJakruvgw3b14oQgcB4BwYVBBqi62xIvbRzAv7Su9t2jK6OR2z3sM/hLJRqIJ5oILMORa7XqrQ=="
```
Comment (dduportal) [inline locals.tf:52]: => to remove from here (not duplicated value)

**Pull request merged**